ilastik/applets/layerViewer/layerViewerGui.py

```diff
@@ -551,6 +552,35 @@ def toggle_show_raw(self, raw_layer_name="Raw Data"):
             for layer in self.layerstack:
                 layer.visible = False
             self.getLayerByName(raw_layer_name).visible = True
+    
+    def cycle_layer_visibility(self):
+        """
+        Cycle between:
+        0: Raw only
+        1: Raw + Uncertainty
+        2: Raw + Probabilities (all channels)
+        """
+        # Hide all layers
+        for layer in self.layerstack:
```

Comment:
Not sure if this is worth it, but updating the visibility property only once avoids emitting unnecessary signals.

```python
for layer in self.layerstack:
    layer.visible = (
        layer.name == "Raw Data" or
        self._layer_toggle_mode == 1 and layer.name == "Uncertainty" or
        self._layer_toggle_mode == 2 and "Probabilities" in layer.name
    )
```